src/eos-google-chrome-installer

```diff
@@ -0,0 +1,218 @@
+        if self._is_connected_state():
+            logging.info("Network connected")
+            return
+
+        logging.info("Not connected to any network, wait for connection")
+
+        loop = GLib.MainLoop()
+
+        def handler(*args):
+            if len(args) < 1:
+                return
+            state = args[0]
+            if not isinstance(state, dbus.Dictionary):
+                return
+            if state.get("Connectivity", None) != NetworkManager.ConnectivityState.FULL:
+                logging.info("Not connected to the internet, still waiting")
+                return
+            if state.get("State", None) != NetworkManager.State.CONNECTED_GLOBAL:
+                logging.info("Not connected to the network, still waiting")
+                return
+
+            logging.info("Connected to the network and the internet")
+            loop.quit()
+
+        dbus_glib.DBusGMainLoop(set_as_default=True)
+        bus = dbus.SystemBus()
+        bus.add_signal_receiver(handler, dbus_interface="org.freedesktop.NetworkManager",
+                                signal_name="PropertiesChanged")
+
+        loop.run()
+
+    def _wait_for_installation(self):
+        loop = GLib.MainLoop()
+
+        def handler(*args):
+            for arg in args[0]:
+                if arg == dbus.String("google-chrome.desktop"):
+                    loop.quit()
+                    return
+            exit_with_error("Gnome Software installed some new application but not Google Chrome. "
+                            "Most probably, user decided to install something else. Quitting...")
+
+        dbus_glib.DBusGMainLoop(set_as_default=True)
+        bus = dbus.SessionBus()
+        bus.add_signal_receiver(handler, dbus_interface="org.gnome.Shell.AppStore",
+                                signal_name="ApplicationsChanged")
+
+        loop.run()
+
+    def _set_as_default_browser(self):
+        mimetypes = ["text/html",
+                     "x-scheme-handler/http",
+                     "x-scheme-handler/https",
+                     "x-scheme-handler/about"]
+        for mimetype in mimetypes:
+            try:
+                subprocess.Popen(["xdg-mime", "default", "google-chrome.desktop", mimetype])
+            except OSError:
+                exit_with_error("Couldn't start xdg-mime. Are xdg-utils installed?")
+        try:
+            subprocess.Popen(["xdg-settings", "set", "default-web-browser", "google-chrome.desktop"])
+        except OSError:
+            exit_with_error("Couldn't start xdg-settings. Are xdg-utils installed?")
+
+    def _touch_done_file(self):
+        chrome_setup_done_path = pathlib.Path(os.path.expanduser("~/.config/google-chrome-initial-setup-done"))
+        if not chrome_setup_done_path.exists():
+            chrome_setup_done_path.touch()
+
+    def _post_install_chrome(self):
+        self._set_as_default_browser()
+        self._touch_done_file()
```

Comment:
Preventing automatic install trigger after the first installation is already done by the file. As far as I know, there is no option in xdg autostart to check gsetting (please correct me if I'm wrong), so assuming that we would only rely on gsetting, we would need to run this script on every boot.